confgenerator/logging_receivers.go

```diff
@@ -152,6 +152,53 @@ func init() {
 	loggingReceiverTypes.registerType(func() component { return &LoggingReceiverSyslog{} })
 }
 
+// A LoggingReceiverTCP represents the configuration for a TCP receiver.
+type LoggingReceiverTCP struct {
+	ConfigComponent `yaml:",inline"`
+
+	ListenHost string `yaml:"listen_host,omitempty" validate:"omitempty,ip"`
+	ListenPort uint16 `yaml:"listen_port,omitempty"`
+}
+
+func (r LoggingReceiverTCP) Type() string {
+	return "tcp"
+}
+
+func (r LoggingReceiverTCP) Components(tag string) []fluentbit.Component {
+	if r.ListenHost == "" {
+		r.ListenHost = "127.0.0.1"
+	}
+	if r.ListenPort == 0 {
+		r.ListenPort = 5170
+	}
+
+	return []fluentbit.Component{{
+		Kind: "INPUT",
+		Config: map[string]string{
+			// https://docs.fluentbit.io/manual/pipeline/inputs/tcp
+			"Name":   "tcp",
+			"Tag":    tag,
+			"Listen": r.ListenHost,
+			"Port":   fmt.Sprintf("%d", r.ListenPort),
+			"Format": "json",
```

Comment:
Just for the record.

There was some discussion about defaulting to `json` or `new-line split format`. Spot checking Fluentd users' tcp plugin configuration, `format json` seems a bit more popular than `format none`: https://github.com/fluent/fluentd/issues?q=is%3Aissue+%22%40type+tcp%22+

I think it's fine to default to `json` for now, and add a flag for users to control the format later if users need that feature.